### PR TITLE
Icon: removed @ebay/skin/icon/common

### PR DESCRIPTION
**Pull request by ianmcburnie**
# Related Issue #540 

This change completely removes the concept of "common" properties shared between background and foreground icons. In theory it was a nice idea, but in practice it led to a few issues. Now background and foreground are completely separate, self-contained modules.

### CODE REVIEW

File: src/less/icon/background/ds6/background.less [hunk 1]
```diff
@@ -70,106 +93,132 @@ span {
...
         .background-image-base64(@ds6-icon-fast-n-free-base64);
+        .icon-size-mixin('icon-fast-n-free');
     }
 
     &.icon--filter-gallery {
         .background-image-base64(@ds6-icon-filter-gallery-base64);
+        .icon-size-mixin('icon-filter-gallery');
     }
 
     &.icon--filter-single {
```
Comment (seangates) [inline src/less/icon/background/ds6/background.less]: Did you miss one?

File: src/less/icon/background/ds6/background.less [hunk 2]
```diff
@@ -178,125 +227,156 @@ span {
 
     &.icon--filter-list {
         .background-image-base64(@ds6-icon-filter-list-base64);
+        .icon-size-mixin('icon-filter-single');
```
Comment (seangates) [inline src/less/icon/background/ds6/background.less]: Should be `icon-filter-list`, I believe.

File: src/less/icon/background/ds6/background.less [hunk 3]
```diff
@@ -1,67 +1,90 @@
...
         .background-image-base64(@ds6-icon-radio-checked-base64);
+        .icon-size-mixin('icon-radio-checked');
     }
 
     &.icon--tick {
         .background-image-base64(@ds6-icon-tick-base64);
+        .icon-size-mixin('icon-tick');
     }
 
     &.icon--menu {
```
Comment (seangates) [inline src/less/icon/background/ds6/background.less]: Missed this one?

Comment (ianmcburnie) [reply src/less/icon/background/ds6/background.less]: Thanks. I've fixed and pushed those 3.

**Pull request merged**